ui/src/App/Main.jsx

```diff
@@ -73,6 +75,18 @@ export default class Main extends Component {
     ) {
       onUnauthorize();
     }
+
+    // Users who were logged in earlier manually will be logged out in order to
+    // be able to complete the third party login flow.
+    if (user && user.identityProviderId === 'manual' && thirdPartyLogin) {
+      onUnauthorize();
+    }
+
+    if (user && thirdPartyLogin) {
+      window.location.href = `${window.location.origin}/login/oauth/authorize${
+        window.location.search
+      }`;
+    }
```

Comment:
If a third party tries to login but the user is not logged in on the site they will be prompted to login. Once logged in, we need to re-initiate the oauth2 login flow since we are dealing with a single page application. I can add a comment to make the intention clearer.